System.ServiceModel.Http 4.6.0-preview6.final incompatible with 4.5.3?

When using a Standard 2.0 library that consumes System.ServiceModel.Http v4.5.3 in an assembly that itself consumes v4.6.0-preview6.final, the compiler shows errors such as:
```
error CS0012: The type 'MessageEncoder' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.ServiceModel.Primitives, Version=4.5.0.3, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
error CS0012: The type 'Binding' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.ServiceModel.Primitives, Version=4.5.0.3, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
```

This appears to be similar to issue https://github.com/dotnet/wcf/issues/2791

Using sn, I can see that the hashes have changed so this may be causing the issue:
```
D:\src\jrinker03\SoapCore>sn -Tp "C:\Users\james\.nuget\packages\system.servicemodel.http\4.5.3\lib\netstandard2.0\System.ServiceModel.Http.dll"

Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Public key (hash algorithm: sha1):
002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9
f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad2361321
02900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93
c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc09334
4d5ad293

Public key token is b03f5f7f11d50a3a

D:\src\jrinker03\SoapCore>sn -Tp "C:\Users\james\.nuget\packages\system.servicemodel.http\4.6.0-preview6.final\lib\netstandard2.0\System.ServiceModel.Http.dll"

Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Public key (hash algorithm: sha1):
0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67
871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0b
d333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307
e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c3
08055da9

Public key token is 31bf3856ad364e35
```

I don't see any notes in the release notes for 4.6.0 Preview5 and Preview6 that there is an expected breaking change here.

Is this similar to #2791 where the key was changed?

--- a/src/System.Private.ServiceModel/src/System.Private.ServiceModel.csproj
+++ b/src/System.Private.ServiceModel/src/System.Private.ServiceModel.csproj
@@ -2,6 +2,7 @@
 
   <PropertyGroup>
     <AssemblyVersion>4.6.0.0</AssemblyVersion>
+    <StrongNameKeyId>Microsoft</StrongNameKeyId>
     <AssemblyName>System.Private.ServiceModel</AssemblyName>
     <RootNamespace>System.ServiceModel</RootNamespace>
     <!-- https://docs.microsoft.com/en-us/nuget/reference/target-frameworks -->

--- a/src/System.ServiceModel.Duplex/src/System.ServiceModel.Duplex.Facade.csproj
+++ b/src/System.ServiceModel.Duplex/src/System.ServiceModel.Duplex.Facade.csproj
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AssemblyVersion>4.6.0.0</AssemblyVersion>
+    <StrongNameKeyId>Microsoft</StrongNameKeyId>
     <AssemblyName>System.ServiceModel.Duplex</AssemblyName>
     <CLSCompliant>true</CLSCompliant>
     <IsPackable>true</IsPackable>

--- a/src/System.ServiceModel.Http/src/System.ServiceModel.Http.Facade.csproj
+++ b/src/System.ServiceModel.Http/src/System.ServiceModel.Http.Facade.csproj
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AssemblyVersion>4.6.0.0</AssemblyVersion>
+    <StrongNameKeyId>Microsoft</StrongNameKeyId>
     <AssemblyName>System.ServiceModel.Http</AssemblyName>
     <CLSCompliant>true</CLSCompliant>
     <IsPackable>true</IsPackable>

--- a/src/System.ServiceModel.NetTcp/src/System.ServiceModel.NetTcp.Facade.csproj
+++ b/src/System.ServiceModel.NetTcp/src/System.ServiceModel.NetTcp.Facade.csproj
@@ -1,6 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AssemblyVersion>4.6.0.0</AssemblyVersion>
+    <StrongNameKeyId>Microsoft</StrongNameKeyId>
     <AssemblyName>System.ServiceModel.NetTcp</AssemblyName>
     <CLSCompliant>true</CLSCompliant>
     <IsPackable>true</IsPackable>

--- a/src/System.ServiceModel.Primitives/src/System.ServiceModel.Primitives.Facade.csproj
+++ b/src/System.ServiceModel.Primitives/src/System.ServiceModel.Primitives.Facade.csproj
@@ -1,6 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AssemblyVersion>4.6.0.0</AssemblyVersion>
+    <StrongNameKeyId>Microsoft</StrongNameKeyId>
     <AssemblyName>System.ServiceModel.Primitives</AssemblyName>
     <CLSCompliant>true</CLSCompliant>
     <IsPackable>true</IsPackable>

--- a/src/System.ServiceModel.Security/src/System.ServiceModel.Security.Facade.csproj
+++ b/src/System.ServiceModel.Security/src/System.ServiceModel.Security.Facade.csproj
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AssemblyVersion>4.6.0.0</AssemblyVersion>
+    <StrongNameKeyId>Microsoft</StrongNameKeyId>
     <AssemblyName>System.ServiceModel.Security</AssemblyName>
     <CLSCompliant>true</CLSCompliant>
     <IsPackable>true</IsPackable>

